validator/sawtooth_validator/gossip/structure_verifier.py

```diff
@@ -72,10 +73,33 @@ def is_valid_batch(batch):
                          "order of transactions in the batch: %s txn: %s",
                          batch.header_signature, header_txn_id)
             return False
+        txn_header = _deserialize_txn_header(txn.header)
+        for dep in txn_header.dependencies:
+            if not dep:
```

Comment:
How about also verifying if the id here is not just random string but has right set of characters we expect in a signature?